### DISCUSSION TITLE
Proxy without objectPath?

### DISCUSSION THREAD

**Discussion by shamlian** [Kistler-Group/sdbus-cpp#258] [Q&A]
I am trying to catch Bluetooth Low Energy connections and disconnections from known  (meaning, the system has seen them) devices in BlueZ. Normally, this requires looking for a PropertiesChanged signal emitted with the Connected property set to either True or False. This would be relatively simple, except that I would need to keep track of the MAC address of all known devices, because these signals are emitted from paths that look like /org/bluez/hci0/dev_{MAC address of device}. I could keep a list of all devices on the tree, but I would prefer to be lazy and instead get a hander for all such signals, no matter who is emitting them. In python-dbus, it is possible to set up a handler for the whole bus without specifying a path. Is such a thing possible in sdbus-c++? It seems to me that sdbus::Proxy classes require objectPath in every constructor. Thanks in advance for any guidance you can offer!

**Comment by dleeds-cpi** [comment]
The "D-Bus Proxy" construct captures a unique service and object path, so it would not be appropriate to create one without an object path. However, it definitely would be nice to be able to register a signal handler with an arbitrary match rule from an `IConnection`. We would most likely want a handle object to manage the lifetime of the callback subscription.

**Reply by sangelovic** [reply]
Good point @dleeds-cpi, I agree. I will try to find time and prepare a PR for this. This will help you solve your issue easily @shamlian .

**Reply by shamlian** [reply]
Thank you both for your help and for a potential solution; I'll look forward to a PR and would love to help test once it's been implemented. I appreciate you taking the time to look into it.

**Reply by sangelovic** [reply]
#264 was prepared and in the meantime already merged to master. Do you want to test it, @shamlian?

**Reply by shamlian** [reply]
Yes, I'd like to. At the moment my time has been preempted but it's on my list. Sorry for the delay and thank you for such rapid help.
